src/Simple.OData.Client.V4.Adapter/CommandFormatter.cs

```diff
@@ -106,19 +98,52 @@ protected override void FormatInlineCount(IList<string> commandClauses)
                 clauses.Add($"{ODataLiteral.Levels}={ODataLiteral.Max}");
             }
 
-            if (selectColumns.Any())
+            if (associationName != StarString)
             {
-                var columns = string.Join(",", SelectPathSegmentColumns(selectColumns, null, entityCollection));
-                if (!string.IsNullOrEmpty(columns))
-                    clauses.Add($"{ODataLiteral.Select}={columns}");
-            }
+                if (expandsToCollection && !ReferenceEquals(association.FilterExpression, null))
+                {
+                    var associatedEntityCollection = _session.Metadata.GetEntityCollection(
+                        _session.Metadata.GetNavigationPropertyPartnerTypeName(entityCollection.Name, associationName));
+                    clauses.Add(
+                        $"{ODataLiteral.Filter}={EscapeUnescapedString(association.FilterExpression.Format(new ExpressionContext(_session, associatedEntityCollection, null, command.DynamicPropertiesContainerName)))}");
+                }
 
-            if (expandsToCollection && orderbyColumns.Any())
-            {
-                var columns = string.Join(",", SelectPathSegmentColumns(orderbyColumns, null, entityCollection)
-                    .Select(x => x.Key + (x.Value ? " desc" : string.Empty)).ToList());
-                if (!string.IsNullOrEmpty(columns))
-                    clauses.Add($"{ODataLiteral.OrderBy}={columns}");
+                if (association.ExpandAssociations.Any())
+                {
+                    var associatedEntityCollection = _session.Metadata.GetEntityCollection(
+                        _session.Metadata.GetNavigationPropertyPartnerTypeName(entityCollection.Name, associationName));
+                    var expandAll = association.ExpandAssociations.FirstOrDefault(a => a.Name == StarString);
+                    if (expandAll != null)
+                    {
+                        clauses.Add($"{ODataLiteral.Expand}=*");
+                    }
+                    else
+                    {
+                        clauses.AddRange(association.ExpandAssociations
+                            .Where(
+                                a => _session.Metadata.HasNavigationProperty(associatedEntityCollection.Name, a.Name))
+                            .Select(a =>
+                                FormatExpansionSegment(a, associatedEntityCollection, ODataExpandOptions.ByValue(),
+                                    command,
+                                    false))
+                            .Select(formattedExpand => $"{ODataLiteral.Expand}={formattedExpand}"));
+                    }
+
+                    var selectColumns = string.Join(",", association.ExpandAssociations.ToList()
```

Comment:
Is the embedded `ToList` required here? `String.Join` will execute this LINQ expression immediately and should not require coercion.

Reply:
You're right, the .ToList() call is not necessary here. I will remove this call.